src/socket.rs

```diff
@@ -1391,6 +1397,10 @@ mod test {
         ($e:expr) => (match $e { Ok(e) => e, Err(e) => panic!("{:?}", e) })
     }
 
+    macro_rules! mutetry {
```

Comment:
It is OK as a temporary solution, but now test machines won't report failures if one of the functions wrapped in `mutetry` fails (as test machines don't parse standard output).